### PR TITLE
feat: Support enabling tests for unmanaged folder

### LINKED ISSUE #1344: Explore better Junit testing experience (brainstorm)
Theme: Testing

### PR DESCRIPTION
**Pull request by jdneo**
This PR fixes #1344. 

We will help users to download test dependencies and setup their projects. (when its a unmanaged folder)

Signed-off-by: Sheng Chen <sheche@microsoft.com>

### CODE REVIEW

File: src/commands/testDependenciesCommands.ts [hunk 1]
```diff
@@ -0,0 +1,324 @@
...
+async function getLibFolder(projectUri: Uri): Promise<string> {
+    const referencedLibraries: any = workspace.getConfiguration('java', projectUri).get('project.referencedLibraries');
+    if (_.isArray(referencedLibraries)) {
+        // do a simple check if the project uses default lib location.
+        if (referencedLibraries.includes('lib/**/*.jar')) {
+            return path.join(projectUri.fsPath, 'lib');
+        }
+    }
+
+    for (let i: number = 0; i < 100; i++) {
```
Comment (Eskibear) [inline src/commands/testDependenciesCommands.ts:123]: what's this `100`?

Comment (jdneo) [reply src/commands/testDependenciesCommands.ts:123]: It's used to get a new available folder. For example, if `test-lib` folder already exists, then we will create `test-lib1`

Comment (jdneo) [reply src/commands/testDependenciesCommands.ts:123]: It's used to create a new available folder. For example. if `test-lib` already exists, then we will use `test-lib1`

Comment (Eskibear) [reply src/commands/testDependenciesCommands.ts:123]: ok, I think it works for most cases.

**Review by Eskibear — APPROVED** [review]

**Pull request merged**